wgsl/index.bs

```diff
@@ -3614,8 +3606,8 @@ is terminated.
 Otherwise, evaluation continues with the next expression or statement after
 the evaluation of the call site of the current function invocation.
 
-If the return type of the function is the void type, then the return statement is
-optional. If the return statement is provided for a void function it must not have an expression.
+If the function doesn't have the return type, then the return statement is
+optional. If the return statement is provided for such a function, it must not have an expression.
```

Comment:
nit: maybe just `If a return statement is provided, it must be a type` ? the `not have an expression` bit is weird here.